pulser-core/pulser/sequence/sequence.py

```diff
@@ -159,6 +160,22 @@ def _slm_mask_time(self) -> list[int]:
             else self._schedule.find_slm_mask_times()
         )
 
+    @property
+    def _in_ising(self) -> bool:
+        return self._in_ising_value
+
+    @_in_ising.setter
+    def _in_ising(self, value: bool) -> None:
+        if not isinstance(value, bool):
+            raise TypeError("_in_ising must be a bool.")
+        if self._in_ising and not value:
+            raise ValueError("Cannot quit ising.")
+        if self._in_xy and value:
+            raise ValueError("Cannot be in ising if in xy.")
+        if self._slm_mask_dmm and self._in_ising != value:
+            self._in_ising_value = value
+            self._set_slm_mask_dmm()
```

Comment:
This fails to set `_in_ising_value = True` when `self._slm_mask_dmm` is not defined. Here's an alternative formulation:
```suggestion
        if self._in_ising == value:
            # If the value doesn't change, do nothing
            return 
        if self._in_ising:    # ie value = False
            # Trying to switch off ising
            raise ValueError("Cannot quit ising.")
        # At this point, value = True
        if self._in_xy:
            raise ValueError("Cannot be in ising if in xy.")
        self._in_ising_value = True
        if self._slm_mask_dmm:
            self._set_slm_mask_dmm()
```

Reply:
I was closing in on this issue as well, that is emerging when setting `self._slm_mask__dmm` after the tests... Your proposition solves it well, I am going for it ;)